geoh5py/groups/interpretation_section.py

```diff
@@ -0,0 +1,312 @@
+#  (at your option) any later version.
+#
+#  geoh5py is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public License
+#  along with geoh5py.  If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+from collections.abc import Mapping
+from typing import TYPE_CHECKING, Any
+from uuid import UUID
+
+from pydantic import BaseModel, Field
+
+from ..shared.utils import str2uuid, to_tuple
+from .base import Group
+
+
+if TYPE_CHECKING:  # pragma: no cover
+    from ..objects import Curve, ObjectBase, Slicer
+
+
+class InterpretationSectionParams(BaseModel):
+    """
+    The class to store the interpretation section.
+
+    :param normal_x: The normal x component.
+    :param normal_y: The normal y component.
+    :param normal_z: The normal z component.
+    :param position_x: The position x component.
+    :param position_y: The position y component.
+    :param position_z: The position z component.
+    """
+
+    normal_x: float = Field(alias="Normal X")
+    normal_y: float = Field(alias="Normal Y")
+    normal_z: float = Field(alias="Normal Z")
+    position_x: float = Field(alias="Position X")
+    position_y: float = Field(alias="Position Y")
+    position_z: float = Field(alias="Position Z")
+
+    @classmethod
+    def create_section(
+        cls,
+        params: dict | InterpretationSectionParams,
+    ) -> InterpretationSectionParams:
+        """
+        Verify/Create the interpretation section from a mapper.
+
+        :param params: The parameters to verify.
+
+        :return: An instance of the class.
+        """
+        if isinstance(params, Mapping):
+            return cls(**params)
+        if isinstance(params, InterpretationSectionParams):
+            return params
+        raise TypeError(
+            "Interpretation section must be an InterpretationSectionParams"
+            f"or a dict, {type(params)} provided."
+        )
+
+
+class InterpretationSection(Group):
+    """The type for the basic Container group."""
+
+    _attribute_map = Group._attribute_map.copy()
+    _attribute_map.update(
+        {
+            "Can add group": "can_add_group",
+        }
+    )
+
+    _TYPE_UID = UUID(fields=(0x27EE59E1, 0xB1CE, 0x41EC, 0x8A, 0x86, 0x8BD3D229E198))
+    _default_name = "Interpretation Section"
+
+    def __init__(self, can_add_group: bool = False, **kwargs):
+        self._can_add_group: bool = self._validate_can_add_group(can_add_group)
+
+        super().__init__(**kwargs)
+
+        self._interpretation_curves: tuple[Curve] | tuple = ()
+        self._interpretation_sections: tuple[InterpretationSectionParams] | tuple = ()
+        self._section_object_id: Slicer | None = None
```

Comment:
```suggestion
        self._section_object: Slicer | None = None
```
Makes more sense to just call it `section_object` I think, since not a uid